### PR TITLE
Aw app test

### PR DESCRIPTION
**Pull request by allisonjw**
@Asilo5 
#### What's this PR do?
- adds testing to app component 
#### Where should the reviewer start?
- open App.test.js to make sure all code is being test
#### How should this be manually tested?
- pull down PR into local test branch and run npm test
#### What are the relevant tickets?
- finished testing App for now #35 
#### Changes
- added missing expect line
- changed fav test to only test for character's Date.now id
#### Questions:
- Additional test will possibly have to be added pending completion of project

### CODE REVIEW

File: src/Components/App/App.test.js [hunk 1]
```diff
@@ -1 +1,107 @@
...
+    expect(wrapper.state().userName).toEqual('');
+    expect(wrapper.state().userQuote).toEqual('');
+    expect(wrapper.state().userRank).toEqual('');
+  });
+
+  it('should have a default value of true for isLoading', () => {
+    expect(wrapper.state().isLoading).toEqual(false);
+  });
+
+  it('should set the state of userName, userQuote, userRanking', () => {
```
Comment (Asilo5) [inline src/Components/App/App.test.js:63]: On this one is there suppose to be an expect?

File: src/Components/App/App.test.js [hunk 2]
```diff
@@ -1 +1,107 @@
...
+    wrapper.setState({ userName: 'name', userQuote: 'quote', userRank: 'rank' });
+  });
+
+  // it.skip('should update state with movies when getMovies is called', async () => {
+  //    await wrapper.instance().getMovies();
+  //    expect(getMovies).toHaveBeenCalled();
+  // });
+
+  it('should retrieve the characters', async () => {
```
Comment (Asilo5) [inline src/Components/App/App.test.js:75]: Nice use of async await here!! Killing it!

File: src/Components/App/App.test.js [hunk 3]
```diff
@@ -1 +1,107 @@
...
+      homeWorld: 'Alderan',
+      population: '200000000000',
+      species: 'Human',
+      films: [
+        "The Empire Strikes Back",
+         "Revenge of the Sith",
+         "Return of the Jedi"
+      ]
+    }]; 
+    it('should toggle favourite characters based on an id', () => {
```
Comment (Asilo5) [inline src/Components/App/App.test.js]: I would word this one different as in the mock there is no ID

File: src/Components/App/App.test.js [hunk 4]
```diff
@@ -1 +1,107 @@
...
+      expect(wrapper.state('favourites')).toEqual([])
+      wrapper.instance().toggleFavourites(character);
+      expect(wrapper.state('favourites')).toEqual([character])
+    });
+
+    it('should be able to add a favourite character to state based on an it', () => {
```
Comment (Asilo5) [inline src/Components/App/App.test.js]: I would also change it to be as similar as possible as to what we have in the original method, an ID is getting passed by not a character.. we will later find the character with that ID when we need to reappend in Favourites components

**Review by Asilo5 — CHANGES_REQUESTED** [review]
I would change some of the favourites testing to accept IDs rather than a whole character since in the App component we are just passing the ID of that character.

File: src/Components/App/App.test.js (hunk 3 above)
Comment (allisonjw) [reply src/Components/App/App.test.js]: @Asilo5 
I made the change for the fav testing that we are only checking for the date.now id in the favourites array.

**Comment by allisonjw** [comment]
@Asilo5 
Ok, I've made the changes and this should be ready for your re-review!

**Review by Asilo5 — APPROVED** [review]

**Comment by Asilo5** [comment]
Everything looks great! Yes!

**Pull request merged**